lib/streamlit/source_util.py

```diff
@@ -29,3 +33,69 @@ def open_python_file(filename):
         return tokenize.open(filename)
     else:
         return open(filename, "r", encoding="utf-8")
+
+
+PAGE_FILENAME_REGEX = re.compile(r"([0-9]*)[_ -]*(.*)\.py")
+
+
+def page_sort_key(script_path: Path) -> Tuple[float, str]:
+    matches = re.findall(PAGE_FILENAME_REGEX, script_path.name)
+
+    # Failing this assert should only be possible if script_path isn't a Python
+    # file, which should never happen.
+    assert len(matches) > 0, f"{script_path} is not a Python file"
```

Comment:
nit: In case, this is something that eventually should be shown to the user, raising an exception here instead of using assert might be slightly better. But since it anyways should never happen, `assert` should be fine here as well.

Reply:
In this case, I don't believe it's possible for a user to run into this unless we mess up something on our end (we use a glob to search for the `.py` files that we pass to the sort function), so leave this as an assert indefinitely is likely fine.